pages/search/index.tsx

```diff
@@ -237,6 +239,22 @@ export async function getServerSideProps({ resolvedUrl, req, query }) {
   const bannerNotification = process.env.SEARCH_RESULTS_NOTIFICATION || ""
   const patronTokenResponse = await initializePatronTokenAuth(req.cookies)
   const results = await fetchResults(mapQueryToSearchParams(query))
+
+  if (
+    !(results instanceof Error) &&
+    hasOneResult(results) &&
+    query.oclc &&
+    query.redirectOnMatch
+  ) {
+    const matchedBib = results.results.itemListElement[0].result
+    return {
+      redirect: {
+        destination: `/bib/${matchedBib.uri}`,
+        permanent: false,
+      },
+    }
+  }
+
```

Comment:
This is the section I'm least sure about. Not sure if there's a more concise way to trigger this branching behavior - both in how the clauses are checked and how the action is carried out

Reply:
I was thinking it might make this cleaner if you move those 4 checks into one util method. This is how we're doing redirecting in My Account - checking for some url params and cookie values and conditionally returning a redirect. Seems ok to me!

Reply:
I moved the check into `searchUtils`